src/chains/filecoin/filecoin/src/blockchain.ts

```diff
@@ -772,7 +773,10 @@ export default class Blockchain extends Emittery.Typed<
 
     // Because we're not cryptographically valid, let's
     // register the deal with the newly created CID
-    this.dealsByCid[proposalCid.value] = deal;
+    this.dealsByCid.set(proposalCid.value, deal);
+
+    // Lets keep deals by id for easy paginated lookup from Ganache UI
+    this.dealsById.set(deal.dealId, deal);
```

Comment:
Are these ever cleaned up? I think these will need to be backed by a DB. :-/

Reply:
No they are not, but #849 replaces them with the appropriate logic when it all gets backed by the DB in that PR